python/ray/data/aggregate.py

```diff
@@ -18,31 +18,49 @@
 
-    Aggregates a collection of inputs of type T into
-    a single output value of type U.
-    See https://www.sigops.org/s/conferences/sosp/2009/papers/yu-sosp09.pdf
-    for more details about accumulator-based aggregation.
+    `AggregateFn` instances are passed to a Dataset's ``.aggregate(...)`` method to
+    specify the steps required to transform and combine rows sharing the same key.
+    This enables implementing custom aggregators beyond the standard
+    built-in options like Sum, Min, Max, Mean, etc.
 
     Args:
-        init: This is called once for each group to return the empty accumulator.
-            For example, an empty accumulator for a sum would be 0.
-        merge: This may be called multiple times, each time to merge
-            two accumulators into one.
-        name: The name of the aggregation. This will be used as the column name
-            in the output Dataset.
-        accumulate_row: This is called once per row of the same group.
-            This combines the accumulator and the row, returns the updated
-            accumulator. Exactly one of accumulate_row and accumulate_block must
-            be provided.
-        accumulate_block: This is used to calculate the aggregation for a
-            single block, and is vectorized alternative to accumulate_row. This will
-            be given a base accumulator and the entire block, allowing for
-            vectorized accumulation of the block. Exactly one of accumulate_row and
-            accumulate_block must be provided.
-        finalize: This is called once to compute the final aggregation
-            result from the fully merged accumulator.
+        init: Function that creates an initial aggregator for each group. Receives a key
+            (the group key) and returns the initial accumulator state (commonly 0,
+            an empty list, or an empty dictionary).
+        accumulate_block: Function that processes an entire block of rows at once. It receives the
+            current accumulator and a block of rows, then returns an updated accumulator.
+            This allows for vectorized operations. Cannot be used if `accumulate_row`
+            is provided.
+        accumulate_row: Function that processes an individual row. It receives the current
+            accumulator and a row, then returns an updated accumulator. Cannot be
+            used if `accumulate_block` is provided.
+        merge: Function that merges two accumulators generated by different workers
+            into one accumulator.
+        finalize: Function that finishes the aggregation by transforming the final
+            accumulator state into the desired output. For example, if your
+            accumulator is a list of items, you may want to compute a statistic
+            from the list. If not provided, the final accumulator state is returned
+            as-is.
+        name: An optional display name for the aggregator. Useful for debugging.
```

Comment:
i think the order should also align with the actual parameters

Reply:
good idea; done